### PR TITLE
update rand to 0.7 (breaking change)

### PR DESCRIPTION
**Pull request by niklasad1**
Updates a bunch of crates to use `rand 0.7`.

It is needed by paritytech/parity-ethereum#11022 but I don't know if it makes sense anymore see RustSec/advisory-db#149 for details.

We could put it on ice if it is too messy

### CODE REVIEW

**Comment by dvdplm** [comment]
What is the rationale for `0.7`?

**Comment by niklasad1** [comment]
It is needed by https://github.com/paritytech/parity-ethereum/pull/11022 but I don't know if it makes sense anymore see https://github.com/RustSec/advisory-db/pull/149 for details.

We could put it on ice if it is too messy

File: fixed-hash/src/hash.rs [hunk 1]
```diff
@@ -512,7 +512,7 @@ macro_rules! impl_rand_for_fixed_hash {
 
 			/// Assign `self` to a cryptographically random value.
 			pub fn randomize(&mut self) {
-				let mut rng = $crate::rand::rngs::EntropyRng::new();
+				let mut rng = $crate::rand::rngs::OsRng;
```
Comment (ordian) [inline fixed-hash/src/hash.rs]: does it work for wasm? see #130 for the previous switch

File: kvdb-rocksdb/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "kvdb-rocksdb"
-version = "0.1.4"
+version = "0.2.4"
```
Comment (ordian) [inline kvdb-rocksdb/Cargo.toml]: why a major bump? ethereum-types is a dev-dependency

File: keccak-hash/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,7 @@ edition = "2018"
 
 [dependencies]
 tiny-keccak = "1.4"
-primitive-types = { path = "../primitive-types", version = "0.5", default-features = false }
+primitive-types = { path = "../primitive-types", version = "0.6", default-features = false }
```
Comment (ordian) [inline keccak-hash/Cargo.toml]: keccak-hash's version needs to be bumped as well

File: fixed-hash/src/hash.rs (hunk 1 above)
Comment (niklasad1) [reply fixed-hash/src/hash.rs]: thanks, probably not then

let me check

File: kvdb-rocksdb/Cargo.toml (hunk 2 above)
Comment (niklasad1) [reply kvdb-rocksdb/Cargo.toml]: mybad

File: keccak-hash/Cargo.toml (hunk 3 above)
Comment (niklasad1) [reply keccak-hash/Cargo.toml]: mybad

File: fixed-hash/src/hash.rs (hunk 1 above)
Comment (niklasad1) [reply fixed-hash/src/hash.rs]: Yes, it works

`$cargo build --target wasm32-unknown-unknown` builds fine for `fixed-hash`

File: contract-address/Cargo.toml [hunk 4]
```diff
@@ -11,9 +11,9 @@ edition = "2018"
 readme = "README.md"
 
 [dependencies]
-ethereum-types = { version = "0.7", path = "../ethereum-types" }
-rlp = { version = "0.4", path = "../rlp" }
-keccak-hash = { version = "0.3", path = "../keccak-hash", default-features = false }
+ethereum-types = { version = "0.8", path = "../ethereum-types" }
```
Comment (ordian) [inline contract-address/Cargo.toml]: we would need to bump `contract-address` to 0.3 too

File: rlp/Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "rlp"
-version = "0.4.2"
+version = "0.5.0"
```
Comment (ordian) [inline rlp/Cargo.toml]: sorry, didn't spot this, but no need for a bump

Comment (niklasad1) [reply rlp/Cargo.toml]: damn, never do this late.... nice catch

**Comment by dvdplm** [comment]
Published a few new versions:

```
* [new tag]             ethbloom-v0.8.0 -> ethbloom-v0.8.0
* [new tag]             fixed-hash-v0.5.0 -> fixed-hash-v0.5.0
* [new tag]             primitive-types-v0.6.0 -> primitive-types-v0.6.0
* [new tag]             ethereum-types-v0.8.0 -> ethereum-types-v0.8.0
* [new tag]             keccak-hash-v0.4.0 -> keccak-hash-v0.4.0
```

Still need to publish `contract-address-0.3.0` (cc @ordian)

**Pull request merged**